lm75bd/lm75bd.c

```diff
@@ -26,9 +26,34 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
 }
 
 error_code_t readTempLM75BD(uint8_t devAddr, float *temp) {
-  /* Implement this driver function */
+    error_code_t errCode;  
+
+    // Check if temp pointer is null
+    if (temp == NULL) {
+        return ERR_CODE_INVALID_ARG; 
+    }
+
+    uint8_t buf[2] = {0};   
+    uint8_t pointerRegister = 0x00;  
+
+    // Send a pointer to the temperature register
+    RETURN_IF_ERROR_CODE(i2cSendTo(devAddr, &pointerRegister, sizeof(pointerRegister)));
+
+    // Read from the temperature register
+    RETURN_IF_ERROR_CODE(i2cReceiveFrom(devAddr, buf, sizeof(buf)));
   
-  return ERR_CODE_SUCCESS;
+    // Extract 11-bit data from the 2 bytes received
+    int16_t tempRaw = (buf[0] << 3) | (buf[1] >> 5);
+
+    // Calculate two's complement for negative temperatures
+    if (tempRaw & (1 << 10)) {
+        tempRaw |= 0xF800;
+    }
```

Comment:
Look at the comment I posted to Luke's question this morning. The second solution is the recommended way of doing things